skimage/color/colorconv.py

```diff
@@ -1456,10 +1460,16 @@ def separate_stains(rgb, conv_matrix):
     >>> ihc = data.immunohistochemistry()
     >>> ihc_hdx = separate_stains(ihc, hdx_from_rgb)
     """
+    from ..exposure import rescale_intensity
+
     rgb = _prepare_colorarray(rgb, force_copy=True)
-    rgb += 2
-    stains = np.reshape(-np.log10(rgb), (-1, 3)) @ conv_matrix
-    return np.reshape(stains, rgb.shape)
+    rgb = np.maximum(rgb, 1E-6)  # avoiding log artifacts
```

Comment:
```suggestion
    np.maximum(rgb, 1E-6, out=rgb)  # avoiding log artifacts
```

Reply:
Isn't the explicit version better understandable?

Reply:
giving the out parameter makes this in place. I'm a big fan of doing little things like that for memory management ;)

Reply:
Might not be so significant for 1MP arrays
```
In [16]: %timeit np.maximum(a, 0.01)                                                                                           
1.1 ms ± 8.98 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [17]: %timeit np.maximum(a, 0.01, out=a)                                                                                    
1.08 ms ± 46.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

```

Reply:
While I understand the appeal, I repeat the question :smile: Isn't it more readable?

Reply:
I think you would really want to reconsider when your data becomes 1GB large, but i'll leave that as a *me* problem, instead of a library problem:
```
In [5]: a = np.full((1, 1024**3//8), fill_value=2, dtype='float')                                        

In [6]: %timeit np.maximum(a, 0.01, out=a)                                                               
138 ms ± 1.75 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [7]: %timeit np.maximum(a, 0.01)                                                                      
567 ms ± 198 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

```

Reply:
:joy: You convinced me, Mark.

Reply:
yeah, the only images i care about are 1GB ;)

Reply:
thx